servicetalk-client-api/src/main/java/io/servicetalk/client/api/ServiceDiscoverer.java

```diff
@@ -26,6 +27,10 @@
  * is called that the service discovery system will push data updates or implementations of this interface will poll for
  * data updates. Changes in the available addresses will be communicated via the resulting {@link Publisher}.
  * <p>
+ * Because typically a {@link ServiceDiscoverer} implementation runs in the background and doesn't require many compute
+ * resources, it's recommended (but not required) to run it and deliver updates on a single thread either for all
+ * discoveries or at least for all {@link Subscriber Subscribers} to the same {@link Publisher}.
```

Comment:
```suggestion
 * discoveries or at least for all {@link Subscriber Subscribers} to the same {@link Publisher}. This will make debugging
   easier as discovery events and the logs they generate will be less susceptible to reordering.
```

Reply:
thanks, applied with minor rewording